New package: pioneers-15.6

tfcolin opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES** (x86_64 only)

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**


<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->

#### Local build testing
- I built this PR locally for these architectures:
  - x86_64
  - x86_64-musl(cross-build)
  - aarch64-musl(cross-build)

vincele commented on srcpkgs/pioneers/template:
```diff
@@ -0,0 +1,31 @@
+#distfiles="https://sourceforge.net/projects/pio/files/Source/${pkgname}-${version}.tar.gz"
+#checksum=9a358d88548e3866e14c46c2707f66c98f8040a7857d47965e1ed9805aeb631d
+
+do_fetch() {
```
Isn't `distfiles` working ? Why ?

tfcolin replied:
I checked again, the "distfiles" is working now. Some problem occurred about the checksum in my environment and it's solved now. Excuse me for I am not familiar with void-packages. Function do_fetch() and do_extract() can be deleted now.

vincele commented on srcpkgs/pioneers/template:
```diff
@@ -0,0 +1,31 @@
+}
+
+do_extract() {
+	tar -xvf ${XBPS_SRCDISTDIR}/pioneers-15.6.tar.gz --strip-components=1
```
Is it because of this `--strip-components=1` ?

vincele commented:
Also rebase & squash commits

tfcolin commented:
I suddenly found I can not ssh my github account in my location. So could any volunteer continue this work?

vincele commented on srcpkgs/pioneers/template:
```diff
@@ -0,0 +1,31 @@
+version=15.6
+revision=1
+#archs="i686 x86_64"
+#build_wrksrc=
```
OK, I'll see what I can do. Thanks for your work on it !

vincele commented:
I don't know (and what I tried didn't work) how to modify your PR (I've rebased and modified your commits locally).
So I'll propose a new PR myself, trying to keep your authorship, if you're OK with that...

tfcolin commented:
> I don't know (and what I tried didn't work) how to modify your PR (I've rebased and modified your commits locally). So I'll propose a new PR myself, trying to keep your authorship, if you're OK with that...

Thank you very much. You are free to decide either you or me as the author. Both are OK for me. :)
